website/validate-document.js

```diff
@@ -0,0 +1,188 @@
+  } else {
+    errors.push(['warn', `Attribute (height="...") not found in ${imageTag}`])
+  }
+
+  if (!decodingAttributeMatch) {
+    errors.push(['warn', `Attribute (decoding="...") not found in ${imageTag}`])
+  }
+
+  if (!loadingAttributeMatch) {
+    errors.push(['warn', `Attribute (loading="...") not found in ${imageTag}`])
+  }
+
+  return [imageSrc, cssWidth, cssHeight];
+};
+
+const checkImages = async (content) => {
+  const errors = [];
+
+  let i = -1;
+  for (const tag of extractImageData(content)) {
+    i++;
+
+    const isMarkdown = tag.startsWith('![');
+    if (isMarkdown) {
+      // Markdown image detected
+      errors.push(['warn', `Markdown image syntax found (${tag}). Use <img src="..." alt="..." width="..." height="..." /> instead for images.`]);
+      continue;
+    }
+
+    let [imageSrc] = checkForImageAttributes(tag, errors);
+
+    const pathPrefix = "/docs/"
+    if (imageSrc.startsWith(pathPrefix)) {
+      imageSrc = imageSrc.substring(pathPrefix.length - 1);
+    } else if (/^https?:/i.test(imageSrc)) {
+      continue;
+    } else {
+      errors.push(`Invalid image src found in ${tag}`);
+      continue;
+    }
+
+    checkImageNameConvention(imageSrc, errors);
+
+  }
+
+  return errors;
+};
+
+const checkDocumentFile = async (fileFullPath, ignore) => {
+  const filePath = path.relative(__dirname, fileFullPath).replaceAll('\\', '/');
+  console.log(`Checking file: "${filePath}"`);
+
+  const errors = [], warnings = [];
+
+  try {
+    const content = fs.readFileSync(fileFullPath, 'utf-8');
+    console.log('Running image checks...');
+    const imageCheckErrors = await checkImages(content);
+    if (imageCheckErrors.length) {
+      const strict = !ignore.ignores(filePath);
+      for (const error of imageCheckErrors) {
+        const [isWarning, message] = !Array.isArray(error) ? [false, error] : [error[0] === 'warn', error[1]];
+        (!isWarning || strict ? errors : warnings).push(message);
+      }
+    } else {
+      console.log('Image checks passed.')
+    }
+
+  } catch (err) {
+    console.error(`Error reading file "${filePath}" : ${err.message}`);
+    errors.push(`File read error: ${err.message}`);
+  }
+
+  if (warnings.length > 0) {
+    console.warn(`Warnings in file "${filePath}":\n- ${warnings.join('\n- ')}`);
+  }
+
+  if (errors.length > 0) {
+    console.error(`Errors in file "${filePath}":\n- ${errors.join('\n- ')}`);
+    return errors;
+  } else {
+    console.log(`${filePath}: Passed all checks.`);
+    return null;
+  }
+
+
+}
+
+const files = process.argv.slice(2);
+if (files.length === 0) {
+  console.log('No markdown files to check.');
+  process.exit(0);
+}
+
+let allErrors = [];
+const checkFiles = async () => {
+  const ig = ignore().add(fs.readFileSync(path.join(__dirname, '.validationignore'), 'utf-8'));
+  for (const file of files) {
+    if (file.endsWith('.mdx')) {
+      const filePath = path.resolve(__dirname, file);
```

Comment:
The workflow passes file paths to the validation script relative to the root of the repo, we need to account for this.

```suggestion
      const filePath = path.resolve(__dirname, '..', file); // Resolve the path relative to the root of the repo
```